### PR TITLE
feat(): add on() event listener api

### PR DESCRIPTION
**Pull request by manucorporat**

### CODE REVIEW

File: packages/stencil-store/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@stencil/store",
-  "version": "0.2.0",
+  "version": "0.2.1",
```
Comment (Serabe) [inline packages/stencil-store/package.json]: There is a breaking change, I would say this is a 0.3.0

File: packages/stencil-store/src/observable-map.test.ts [hunk 2]
```diff
@@ -32,22 +32,28 @@ describe('reset', () => {
     expect(state).not.toHaveProperty('hola');
   });
 
-  test('calls reset subscriptions (fn)', () => {
-    const { subscribe, reset } = createObservableMap({ hola: 'hola' });
+
+  test('calls reset subscriptions (object)', () => {
```
Comment (Serabe) [inline packages/stencil-store/src/observable-map.test.ts]: Add its own test ('calls onChange'). 
Remove clarification about subscription object, as there is no longer an option for function.

File: packages/stencil-store/src/observable-map.test.ts [hunk 3]
```diff
@@ -79,26 +85,26 @@ describe.each([
       expect(getter(state, get, 'hola')).toBe('ola');
     });
 
-    test('calls subscriptions (fn)', () => {
+    test('calls subscriptions (object)', () => {
```
Comment (Serabe) [inline packages/stencil-store/src/observable-map.test.ts]: Remove `object` as `function` is no longer usable.

File: packages/stencil-store/src/observable-map.test.ts [hunk 4]
```diff
@@ -123,30 +129,42 @@ describe.each([
       expect(state.hola).toBe('ola');
     });
 
-    test('calls subscriptions (fn)', () => {
+    test('calls subscriptions (object)', () => {
```
Comment (Serabe) [inline packages/stencil-store/src/observable-map.test.ts]: Remove _(object)_

File: packages/stencil-store/src/observable-map.ts [hunk 5]
```diff
@@ -38,49 +39,49 @@ export const createObservableMap = <T extends { [key: string]: any }>(
     },
   });
 
-  const subscribe = (subscription: StoreSubscription<T> | StoreSubscriptionObject<T>): void => {
-    subscriptions.push(toSubscription(subscription));
+  const on: OnHandler<T> = (eventName, callback) => {
+    let listeners: any[] = setListeners;
```
Comment (Serabe) [inline packages/stencil-store/src/observable-map.ts]: Can we throw if `eventName` is not a valid one?

File: packages/stencil-store/src/observable-map.ts [hunk 6]
```diff
@@ -38,49 +39,49 @@ export const createObservableMap = <T extends { [key: string]: any }>(
...
+      if (key === propName) {
+        cb(newValue);
+      }
+    });
+    on('reset', () => {
+      cb(defaultState[propName]);
+    });
+  };
+
+  const use = (subscription: StoreSubscriptionObject<T>): void => {
```
Comment (Serabe) [inline packages/stencil-store/src/observable-map.ts]: Can we have multiple subscriptions here?

File: packages/stencil-store/src/observable-map.ts [hunk 7]
```diff
@@ -38,49 +39,49 @@ export const createObservableMap = <T extends { [key: string]: any }>(
...
+    if (subscription.get) {
+      on('get', subscription.get);
+    }
+    if (subscription.reset) {
+      on('reset', subscription.reset);
+    }
   };
 
   return {
-    /**
```
Comment (Serabe) [inline packages/stencil-store/src/observable-map.ts]: Why removing documentation?

File: packages/stencil-store/package.json (hunk 1 above)
Comment (manucorporat) [reply packages/stencil-store/package.json]: i think this ia a previous release, we never updated the package.json my fault

File: packages/stencil-store/src/observable-map.ts (hunk 5 above)
Comment (manucorporat) [reply packages/stencil-store/src/observable-map.ts]: ok

File: packages/stencil-store/src/observable-map.ts (hunk 7 above)
Comment (manucorporat) [reply packages/stencil-store/src/observable-map.ts]: it was not removed, we moved to the actual types, so users can see the docs!

File: packages/stencil-store/src/observable-map.ts (hunk 6 above)
Comment (manucorporat) [reply packages/stencil-store/src/observable-map.ts]: ups

**Pull request merged**